### ISSUE TITLE
With jspm 0.17 beta, 'jspm bundle --inject' arithmetic not working

### ISSUE THREAD

**Issue by kwesterfeld** [jspm/jspm-cli#1578]
With 0.16, I use this to bundle my application dependencies, MINUS my app an its static resources:

```
jspm bundle 'app/**/*.js - [app/**/*.js] - [app/**/*.css!] - [app/**/*.html!text] + css + systemjs-hot-reloader' smc/bundle/smc-dependency-bundle.js --inject;
```

With the 0.17 release, this bundle no contains my application's css and html/text, which makes it difficult to use.  I have to essentially run my application unbundled, causing performance pain.

Did the syntax change perhaps?

**Comment by kwesterfeld** [comment]
I found that I can get the old behavior by using the following, which is really ugly embedding versions of the plugins:
```
jspm bundle 'app/**/*.js - [app/**/*.js] - [app/**/*.css!github:systemjs/plugin-css@0.1.20/css.js] - [app/**/*.html!github:systemjs/plugin-text@0.0.4/text.js] + css + systemjs-hot-reloader' smc/bundle/smc-dependency-bundle.js --inject;
```

**Comment by guybedford** [comment]
Perhaps try removing the `[]` from around the CSS and text requires? Does that help here?

**Comment by kwesterfeld** [comment]
I tried this, no dice (it still pulls app css and html into the injected bundle):

```
jspm bundle 'app/**/*.js - [app/**/*.js] - app/**/*.css! - app/**/*.html!text + css + systemjs-hot-reloader' --inject
```

**Comment by guybedford** [comment]
@kwesterfeld thanks for the quick response. Does `jspm bundle 'app/**/*.css!'` work for you at all then? Otherwise it sounds like this could be some sort of canonicalization bug.

**Comment by kwesterfeld** [comment]
The instructions you gave bundle up the css, yes....gives a nice verbose listing with each file like this:

```
% jspm bundle 'app/**/*.css!'
     Building the bundle tree for app/**/*.css!...

       app/css/bootstrap-overrides.css!github:systemjs/plugin-css@0.1.20
       app/css/codemirror-overrides.css!github:systemjs/plugin-css@0.1.20
       app/css/fonts/fonts.css!github:systemjs/plugin-css@0.1.20

...etc
```

**Comment by rpkilby** [comment]
I have run into this as well (not injecting, doesn't seem relevant?). In the following calls, both the `css` and `html` files are included in the bundle. 
```
jspm bundle 'app/**/* - [app/**/*] - app/**/*.css! - app/**/*.html!text' deps.bundle.js
jspm bundle 'app/**/* - [app/**/*] - [app/**/*.css!] - [app/**/*.html!text'] deps.bundle.js
```

Running `jspm bundle 'app/**/*.css!'` produces expected output.

**Comment by guybedford** [comment]
Thanks, I managed to track this down in https://github.com/systemjs/builder/commit/d3d75889ee5fb92bc3d85f9e9c37b1838e3ed698.

Note this problem goes away if using `loader` configuration instead of plugin syntax as it just becomes `bundle 'app/**/* - app/**/*.css'` sort of exclusion, which is advised.

**Comment by kwesterfeld** [comment]
@guybedford Thanks for the fix!  I did some digging and I cannot find what you mean by this: "`loader` configuration, instead of plugin syntax".   Since you are advising to do that, I can make this type of change easily but am a little lost as to what I should be doing there.

**Comment by guybedford** [comment]
Sure no problem. I mean the approach:

```javascript
meta: {
  "*.css": {
    "loader": "css"
  }
}
```

in the package configuration for the app, so plugin syntax is not used at all and the require just reads `import './file.css'`.

**Comment by kwesterfeld** [comment]
Oh, that's nice.   Thanks again!

**Comment by kwesterfeld** [comment]
Tested this configuration and it's really a lot nicer than the !css and !text syntax.  Can I suggest you make these the defaults or perhaps the first thing someone adds -- linked via docs?

**Comment by rpkilby** [comment]
This is great. Corrected all of the imports and can now bundle the dependencies.
```
jspm bundle app/**/* - [app/**/*] deps.bundle.js
```

**Comment by guybedford** [comment]
Released in 0.17.0-beta.10.
